src/influxdb_ioxd/http.rs

```
@@ -604,6 +605,36 @@ async fn set_writer<M: ConnectionManager + Send + Sync + Debug + 'static>(
     Ok(response)
 }
 
+#[tracing::instrument(level = "debug")]
+async fn get_writer<M: ConnectionManager + Send + Sync + Debug + 'static>(
+    req: Request<Body>,
+) -> Result<Response<Body>, ApplicationError> {
+    let id = {
+        let server = Arc::clone(&req.data::<Arc<AppServer<M>>>().expect("server state"));
+        server.require_id()
+    };
+
+    // Parse the JSON body into a structure
+    #[derive(Serialize)]
+    struct WriterIdBody {
+        id: u32,
+    }
+
+    let body = WriterIdBody {
+        id: id.unwrap_or(0),
```

a nit is that this case should be an error, right? Like when the server doesn't have a write id configured this code will return an id of `0` rather than some sort of error about the id not being set.

---

0 is the not set value on the server side - I agree it should probably return an error, but it's about to be ripped out for gRPC anyway so punted it.

---

👍